apps/system/lockscreen/js/lockscreen.js

```diff
@@ -434,10 +434,14 @@
     // bug 992277. We don't need to use SettingsListener because
     // we're only interested in changes to the setting, and don't
     // keep track of its value.
-    navigator.mozSettings.addObserver('lockscreen.lock-immediately',
-      (function(event) {
-      if (event.settingValue === true) {
+    window.SettingsListener.observe('lockscreen.lock-immediately',
```

Comment:
Please make sure that this is executed only once. This codes lives in the system app, so adding mozSettings observers continuously will leak.